src/data-workspace/data-workspace.js

```diff
@@ -15,11 +15,15 @@ import {
 import styles from './data-workspace.module.css'
 import { EntryForm } from './entry-form.js'
 import { FinalFormWrapper } from './final-form-wrapper.js'
+import { useHasCommentContext } from './has-comment/has-comment-context.js'
 
 export const DataWorkspace = ({ selectionHasNoFormMessage }) => {
     const { data } = useMetadata()
     const [dataSetId] = useDataSetId()
-    const initialDataValuesFetch = useDataValueSet()
+    const { populateHasCommentContextForDataSetValues } = useHasCommentContext()
+    const initialDataValuesFetch = useDataValueSet((data) => {
+        populateHasCommentContextForDataSetValues(data.dataValues)
+    })
```

Comment:
Is this not something we always want to do? Would it not make sense to move this logic to the `useDataValueSet`-hook? 
I guess one downside is that every component that uses that hook would also be subscribed to the context (since you would need `useContext` inside).

On that note, we could also do as we do with some of the other contexts - splitting it into two contexts, so you have one "setter"-context, and one "value"-context. That way this component would not need to update when the value is changed. This can be done regardless if we go the route above.

Could potentially also create a "composing"-hook, like `useDataValueSetWithComment` or something?